src/Registry/TypeRegistry.php

```diff
@@ -1015,18 +1018,25 @@ public function register_fields( string $type_name, array $fields = [] ) {
 	/**
 	 * Add a field to a Type in the Type Registry
 	 *
-	 * @param string $type_name  Name of the type in the Type Registry to add the fields to
-	 * @param string $field_name Name of the field to add to the type
-	 * @param array  $config     Info about the field to register to the type
+	 * @param string $type_name                       Name of the type in the Type Registry to add
+	 *                                                the fields to
+	 * @param string $field_name                      Name of the field to add to the type
+	 * @param array  $config                          Info about the field to register to the type
 	 *
 	 * @return void
+	 * @throws \Exception
 	 */
-	public function register_field( string $type_name, string $field_name, array $config ) {
+	public function register_field( string $type_name, string $field_name, array $config ): void {
+
+
 		add_filter(
 			'graphql_' . $type_name . '_fields',
 			function ( $fields ) use ( $type_name, $field_name, $config ) {
 
-				$field_name = Utils::format_field_name( $field_name );
+				// Whether the field should be allowed to have underscores in the field name
+				$allow_field_underscores = isset( $config['allow_field_underscores'] ) && true === $config['allow_field_underscores'];
```

Comment:
Irony notwithstanding, isnt the convention for `$config` keys themselves to currently use `camelCase`?

Reply:
(same in WPConnectionType and WPMutationType)